test/controllers/order_items_controller_test.rb

```diff
@@ -0,0 +1,164 @@
+require "test_helper"
+describe OrderItemsController do
```

Comment:
As I mention in another comment, there's a logic in this controller that belongs in the model. That said, the testing here is very well-written and thorough!